lib/pbio/drv/usb/stm32_usbd/usbd_desc.c

```diff
@@ -265,4 +410,51 @@ void USBD_Pybricks_Desc_Init(void) {
         USBD_DeviceDesc[11] = HIBYTE(PBDRV_CONFIG_USB_PID_1);
     }
     #endif
+
+    // REVISIT: we could probably move this to the static initializer above,
+    // but it would probably be a bit fragile and ugly
+
+    uint8_t *ptr = &USBD_BOSDesc[USB_SIZ_BOS_DESC_CONST];
+
+    /* Add firmware version */
+    *ptr++ = USB_SIZ_PLATFORM_HDR + sizeof(firmware_version) - 1;
+    *ptr++ = USB_DEVICE_CAPABITY_TYPE;
+    *ptr++ = 0x05;
```

Comment:
Let's add comments here like `/* bDevCapabilityType = PLATFORM */` here similar to how the static arrays are commented above.

Reply:
I think at one point we discussed moving all of this into a const array. That discussion seems to have disappeared, but is that still your preference or do you want to continue with the runtime approach?

Reply:
Unless there is a better place to add the device name, we need to add it here. Since the device name is set when flashing the firmware and not at compile time, then either we have to have trailing null characters in a fixed sized device name or do it at runtime like we are already doing here. So it seems like this is probably the better approach (and we can get rid of the REVISIT comment).